### PR TITLE
chore: remove `foundry-wallets` and `script` cheatcode

### PR DESCRIPTION
**Pull request by agostbiro**
Remove the `foundry-wallets` crate and the `script` cheatcode that depends on it as the former has a dependency that doesn't build on `linux-musl`. The `script` cheatcode is used to customize deployment logic which we don't need.

### CODE REVIEW

**Review by Wodann — COMMENTED** [review]
I have one nice to have comment related to the code changes.

Otherwise I find it hard to make a call on this PR. It seems to me that the removal of cheatcodes is a product decision.

There are several questions that come to mind:
- How much harder does the removal of these cheatcodes make the adoption of EDR's solidity test runner for projects that support Foundry Solidity tests? (measured against our OKR of Solidity Tests being a drop-in replacement)
- How much are the cheatcodes used?
- If used often, to what extend should we have a replacement for this behaviour?
- Can Hardhat fill that gap?

I'd have to cede to @fvictorio and potentially leadership on that topic.

File: .cargo/config.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [alias]
+cheats = "test -p foundry-cheatcodes-spec --features schema tests::"
```
[the remark this answers is not in the document]
Comment (agostbiro) [reply .cargo/config.toml]: I agree this is a bit confusing, bc it's used regenerate a JSON file. I fixed this in [Rename cargo cheats to generate-cheats-interface](https://github.com/NomicFoundation/edr/pull/477/commits/6215c5f42accf2b5d3bad16b09df660ac37db807) and more comprehensively by adding docs in https://github.com/NomicFoundation/edr/pull/478

[the remark this answers is not in the document]
Comment (Wodann) [reply .cargo/config.toml]: Thanks!

**Review by fvictorio — APPROVED** [review]

**Comment by fvictorio** [comment]
> There are several questions that come to mind:
> 
> * How much harder does the removal of these cheatcodes make the adoption of EDR's solidity test runner for projects that support Foundry Solidity tests? (measured against our OKR of Solidity Tests being a drop-in replacement)
> * How much are the cheatcodes used?
> * If used often, to what extend should we have a replacement for this behaviour?
> * Can Hardhat fill that gap?

These are good questions. My understanding is that this PR removes two things:

1) The scripting feature, which makes complete sense, plus the wallets crate that is only used in scripts
2) Cheatcodes that are mainly used in scripts

The second one seems riskier. Maybe some of these cheatcodes _are_ useful in tests in some cases, I don't really know. But this seems like a good bet to me. If it turns out that some of these cheatcodes are needed or really useful, we can selectively add them later.

**Comment by agostbiro** [comment]
> 2. Cheatcodes that are mainly used in scripts
> 
> The second one seems riskier. Maybe some of these cheatcodes _are_ useful in tests in some cases, I don't really know. But this seems like a good bet to me. If it turns out that some of these cheatcodes are needed or really useful, we can selectively add them later.

Thanks for the review! To clarify, only cheatcodes were removed that are exclusively used by scripts, because they depend on the wallet abstraction. Cheatcodes that take a raw key and return a signature remain.

**Comment by Wodann** [comment]
> To clarify, only cheatcodes were removed that are exclusively used by scripts, because they depend on the wallet abstraction. Cheatcodes that take a raw key and return a signature remain

Thanks for clarifying. That was a missing piece for me

**Pull request merged**